Create endpoints to Create, Update, Delete users

pattrickrice opened the pull request:
- Import security module for hashing passwords.
- Created create, update, delete api endpoints.
- Still need orm hookup.

pattrickrice committed:
Create endpoints to Create, Update, Delete users

- Import security module for hashing passwords.
- Created create, update, delete api endpoints.
- Still need orm hookup.

pattrickrice committed:
Add isAdmin boolean to the user model

pattrickrice committed:
Update user API create, update, delete to use headers

anderMatt commented on src/main/java/github/paz/awardportal/controller/UserController.java:
```diff
@@ -57,10 +57,10 @@
-            @RequestParam("lastName") String lastName,
-            @RequestParam("email") String email,
-            @RequestParam("password") String password) {
+            @RequestHeader("firstName") String firstName,
```
Looks good! I'd like to point out another Spring annotation that could potentially be used here, @RequestBody. (https://www.baeldung.com/spring-request-response-body)

Suppose we POST the new user json to this endpoint:

{"firstName": "name", "lastName": "name", "email": "theEmail"...}

The @RequestBody annotation will automatically instantiate a User object from the json, and pass it to the controller method. 
 
createUser(@RequestBody User user) {   //Spring instantiates the user from the json data.
    <do stuff with the User object...hash its password, pass to database code, etc.>
    return ResponseEntity.ok();
}

This is just a suggestion that allows us to take full advantage of some utilities Spring provides, but passing data in headers and instantiating User objects ourselves is perfectly fine too!

anderMatt replied:
Edit: Just noticed first name is a Header and not RequestParam - was this intended or a mistake?

pattrickrice replied:
It was just a mistake.

pattrickrice committed:
Switch to @RequestBody annotation in user controller

- API now uses @Requestbody instead of headers or params
  only in create, update, delete.
- Create BaseUser for instantiating a User object that
  does not yet have an ID.

pattrickrice commented:
Thanks for suggesting the Requestbody annotation, I hadn't seen that yet. This new request implements that.

pattrickrice committed:
Create unit tests for User model.
